source/extensions/filters/http/cache/cache_filter.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/cache/v3alpha/cache.pb.h"
+
+#include "common/common/logger.h"
+
+#include "extensions/filters/http/cache/http_cache.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+/**
+ * A filter that caches responses and attempts to satisfy requests from cache.
+ * It also inherits from std::enable_shared_from_this so it can pass shared_ptrs to async methods,
+ * to ensure that it doesn't get destroyed before they complete.
```

Comment:
This is not something that we typically do, as it makes lifetimes pretty hard to reason about. If we do this, the stream may be reset before before the async completion. I think it would be better to do explicit cancellations of async methods during onDestroy()? This is what we do elsewhere.

Reply:
Done.